app/presenters/collection_details_presenter_umm_json.rb

```diff
@@ -207,11 +207,15 @@ def related_urls(collection)
       end
     end
 
-    # URLs hsould be listed alphabetically and grouped by Type
+    # URLs should be listed alphabetically and grouped by Type
     related_urls.each do |url_category|
       url_category[:urls].sort_by! {|url| [url['Type'], url['Subtype'], url['URL']]}
     end
 
+    # related_urls should be empty if there are no Related URLs
+    related_urls.each {|x| if !(x[:urls].empty?) then return related_urls end}
+    related_urls = []
+
     related_urls
```

Comment:
I like this strategy!

Since you're returning an empty array when related_urls is empty, I wonder if the assignment 'related_urls = []' is even necessary.  Line 216 appears to return related_urls in the case that *any* of the sub arrays are *not* empty - so our 'success' condition is covered.  So, you should be able to delete line 219 (and 218) entirely and change 217 to just return an empty array, such as this:

related_urls.each {|x| if !(x[:urls].empty?) then return related_urls end}
[]

Reply:
This totally makes sense, I agree it should be simplified.